shared/modules/transaction.utils.js

```diff
@@ -131,3 +132,79 @@ export async function determineTransactionType(txParams, query) {
 
   return { type: result, getCodeResponse: contractCode };
 }
+
+const INFERRABLE_TRANSACTION_TYPES = [
+  TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+  TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+  TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
+  TRANSACTION_TYPES.CONTRACT_INTERACTION,
+  TRANSACTION_TYPES.SIMPLE_SEND,
+];
+
+/**
+ * Given a transaction meta object, determine the asset type that the
+ * transaction is dealing with, as well as the standard for the token if it
+ * is a token transaction.
+ *
+ * @param {import('../constants/transaction').TransactionMeta} txMeta -
+ *  transaction meta object
+ * @param {EthQuery} query - EthQuery instance
+ * @param {Function} getTokenStandardAndDetails - function to get token
+ *  standards and details.
+ * @returns {{ assetType: string, tokenStandard: string}}
+ */
+export async function determineTransactionAssetType(
+  txMeta,
+  query,
+  getTokenStandardAndDetails,
+) {
+  // If the transaction type is already one of the inferrable types, then we do
+  // not need to re-establish the type.
+  let inferrableType = txMeta.type;
+  if (INFERRABLE_TRANSACTION_TYPES.includes(txMeta.type) === false) {
+    // Because we will deal with all types of transactions (including swaps)
+    // we want to get an inferrable type of transaction that isn't special cased
+    // that way we can narrow the number of logic gates required.
+    const result = await determineTransactionType(txMeta.txParams, query);
+    inferrableType = result.type;
+  }
+
+  const isTokenMethod = [
+    TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
+  ].find((methodName) => isEqualCaseInsensitive(methodName, inferrableType));
+
+  if (isTokenMethod) {
+    let details;
+    try {
+      details = await getTokenStandardAndDetails(
+        txMeta.txParams.to,
+        txMeta.txParams.from,
+      );
+    } catch {
+      // noop
+    }
+
+    if (details.standard) {
```

Comment:
Since we wont have `details.standard` unless the `try` succeeds, should we move this block into the `try`?  Especially if our catch does nothing?